UpcomingMovies.Core/Services/MovieService.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            var url = BuildUrl($"/movie/{MovieListTypeToString(type)}", parameters);
+
+            var result = new ResponseInfo<MovieListResponse>();
+            try
+            {
+                var response = await _httpClientService.GetAsync(url);
+                if (response.IsSuccessStatusCode)
+                {
+                    var responseString = await response.Content.ReadAsStringAsync();
+                    result.Result = JsonConvert.DeserializeObject<MovieListResponse>(responseString);
+                    result.IsSuccess = true;
+                }
+                else
+                {
+                    result.Error = UiMessages.ServiceUnavailable;
+                }
+            }
+            catch (Exception e)
+            {
+                result.Error = e.Message;
+            }
+
+            return result;
+        }
+
+        public async Task<ResponseInfo<Movie>> GetMovieDetailAsync(string id, Dictionary<string, object> parameters = null)
+        {
+            var url = BuildUrl($"/movie/{id}", parameters);
+            url = Url.Combine(url).SetQueryParam("append_to_response", "videos,credits");
+
+            var result = new ResponseInfo<Movie>();
+            try
+            {
+                var response = await _httpClientService.GetAsync(url);
+                if (response.IsSuccessStatusCode)
+                {
+                    var responseString = await response.Content.ReadAsStringAsync();
+                    result.Result = JsonConvert.DeserializeObject<Movie>(responseString);
+                    result.IsSuccess = true;
+                }
+                else
+                {
+                    result.Error = UiMessages.ServiceUnavailable;
+                }
+            }
+            catch (Exception e)
+            {
+                result.Error = e.Message;
+            }
+
+            return result;
+        }
+
+        public async Task<ResponseInfo<MovieListResponse>> SearchMoviesAsync(Dictionary<string, object> parameters = null)
+        {
+            var url = BuildUrl($"/search/movie", parameters);
+
+            var result = new ResponseInfo<MovieListResponse>();
+            try
+            {
+                var response = await _httpClientService.GetAsync(url);
+                if (response.IsSuccessStatusCode)
+                {
+                    var responseString = await response.Content.ReadAsStringAsync();
+                    result.Result = JsonConvert.DeserializeObject<MovieListResponse>(responseString);
+                    result.IsSuccess = true;
+                }
+                else
+                {
+                    result.Error = UiMessages.ServiceUnavailable;
+                }
+            }
+            catch (Exception e)
+            {
+                result.Error = e.Message;
+            }
+
+            return result;
+        }
+
+        string BuildUrl(string api, Dictionary<string, object> parameters = null)
```

Comment:
I see you prefer not using `private` modifier. Is that a personal preference?

Reply:
Yes it's personal preference but also there was a time where the VS for mac would darken the private modifier and it would say "its redundant to specify it" or something like that. So I stopped writing it. We don't like to write redundant stuff, right?

Reply:
It depends. Usually we do write every modifier just to follow the pattern. It's a matter of taste+company standards though.